src/server/game/Entities/Unit/Unit.h

```diff
@@ -1675,10 +1675,11 @@ class TC_GAME_API Unit : public WorldObject
         void UpdateInterruptMask();
 
         uint32 GetDisplayId() const { return GetUInt32Value(UNIT_FIELD_DISPLAYID); }
-        virtual void SetDisplayId(uint32 modelId);
+        virtual void SetDisplayId(uint32 modelId, float displayScale = 1.f);
         uint32 GetNativeDisplayId() const { return GetUInt32Value(UNIT_FIELD_NATIVEDISPLAYID); }
+        uint32 GetNativeDisplayScale() const { return 1.f/*GetUInt32Value(UNIT_FIELD_NATIVEXDISPLAYSCALE)*/; }
         void RestoreDisplayId(bool ignorePositiveAurasPreventingMounting = false);
-        void SetNativeDisplayId(uint32 modelId) { SetUInt32Value(UNIT_FIELD_NATIVEDISPLAYID, modelId); }
+        void SetNativeDisplayId(uint32 displayId, float displayScale = 1.f) { SetUInt32Value(UNIT_FIELD_NATIVEDISPLAYID, displayId); /*SetUInt32Value(UNIT_FIELD_NATIVEXDISPLAYSCALE, displayScale);*/ }
```

Comment:
```cpp
src/server/game/Entities/Unit/Unit.h:1682:57:
 fatal error: unused parameter 'displayScale' [-Wunused-parameter]
        void SetNativeDisplayId(uint32 displayId, float displayScale = 1.f) { SetUInt32Value(UNIT_FIELD_NATIVEDISPLAYID, displayId); /*SetUInt32Value(UNIT_FIELD_NATIVEXDISPLAYSCALE, displayScale);*/ }
                                                        ^
```

Reply:
It will be used when UpdateFields are updated and code is uncommented